bulbs/contributions/models.py

```diff
@@ -159,14 +160,17 @@ class ContributorRole(Indexable):
     payment_type = models.IntegerField(choices=ROLE_PAYMENT_TYPES, default=MANUAL)
 
     def save(self, *args, **kwargs):
+        created = bool(self.pk is None)
         super(ContributorRole, self).save(*args, **kwargs)
-        if self.payment_type == 1:
-            self.create_feature_type_rates()
-
-    def create_feature_type_rates(self):
-        for feature_type in FeatureType.objects.all():
-            rate = FeatureTypeRate.objects.filter(role=self, feature_type=feature_type)
-            if not rate.exists():
+        self.create_feature_type_rates(created)
+        update_role_rates.delay(self)
```

Comment:
Web consensus is that Passing object is dangerous, PK is better. 
----
https://denibertovic.com/posts/celery-best-practices/
You shouldn't pass Database objects (for instance your User model) to a background task because the serialized object might contain stale data. What you want to do is feed the task the User id and have the task ask the database for a fresh User object.